app/components/gh-nav-menu.js

```diff
@@ -11,9 +11,14 @@ export default Component.extend({
 
     open: false,
 
-    navMenuIcon: computed('config.blogUrl', 'settings.icon', function () {
+    navMenuIcon: computed('config.blogUrl', 'settings.icon', 'ghostPaths.subdir', 'ghostPaths.url', function () {
+        let subdirRegExp = new RegExp(`^${this.get('ghostPaths.subdir')}`);
         let blogIcon = this.get('settings.icon') ? this.get('settings.icon') : 'favicon.ico';
-        let url = `${this.get('config.blogUrl')}/${blogIcon}`;
+        let url;
+
+        blogIcon = blogIcon.replace(subdirRegExp, '');
+
+        url = this.get('ghostPaths.url').join(this.get('config.blogUrl'), blogIcon).replace(/\/$/, '');
```

Comment:
If you add this line after it then it will force a fresh download when the custom icon is removed:

```javascript
url += `?t=${(new Date()).valueOf()}`;
```